lib/datasource/nuget/v3.ts

```diff
@@ -29,97 +39,114 @@ export async function getQueryUrl(url: string): Promise<string | null> {
   }
 
   try {
-    // TODO: fix types
-    const servicesIndexRaw = await http.getJson<any>(url);
-    const searchQueryService = servicesIndexRaw.body.resources.find(
-      (resource) => resource['@type']?.startsWith(resourceType)
+    const servicesIndexRaw = await http.getJson<ServicesIndexRaw>(url);
+    const services = servicesIndexRaw.body.resources.find((resource) =>
+      resource['@type']?.startsWith(resourceType)
     );
```

Comment:
We should make sure to fetch the higest version available, otherwise we miss semver v2 packages

https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource#versioning

We should try to get `RegistrationsBaseUrl/3.6.0`

Reply:
To simple reproduce check [`Microsoft.Extensions.Logging`](https://www.nuget.org/packages/Microsoft.Extensions.Logging)

latest prerelease is `5.0.0-preview.7.20364.11` which is semver 2.

if we use first `RegistrationsBaseUrl` we will use https://api.nuget.org/v3/registration5-semver1/microsoft.extensions.logging/index.json, where the prereleases are missing

if we use `RegistrationsBaseUrl/3.6.0`, we will use https://api.nuget.org/v3/registration5-gz-semver2/microsoft.extensions.logging/index.json, where all versions are available.